### PR TITLE
Rewrite github_tools functions that do not rely on github features

### PR DESCRIPTION
**Pull request by appeltel**
Move functions in github_tools that do not relate to github
specific features to git_tools, and rewrite these functions to
provide (nearly) the same functionality using the GitPython package
rather than the github API and requests.

This allows new releases to be created without a github token or
even a github account. Previously, the only functionality in making a
new release requiring the github API was for `git cirrus release new`
to retrieve all commits since the last release and add records of
these commits to `HISTORY.md`.

The old method for building `HISTORY.md` was apparently not working
correctly, as github was returning the last 30 commits, rather than
all commits since the last release. This bug is fixed with the changes.

Additionally, the old function `get_tags` was sorting the tags in
alphabetical order by tag name rather than by newest first. This would
cause version `0.1.13` to appear as older than `0.1.9`. This bug is
also fixed with the new function.

Unit tests have been rewritten and/or  to reflect the new functions that 
do not use the github API.

### CODE REVIEW

File: src/cirrus/git_tools.py [hunk 1]
```diff
@@ -200,3 +203,145 @@ def get_diff_files(repo_dir):
...
+        row = {
+            'committer' : commit.committer.name,
+            'message' : commit.message,
+            'date' : str(arrow.get(commit.committed_date))
+        }
+        result.append(row)
+    return result
+
+
+def format_commit_messages(rows):
```
Comment (appeltel) [inline src/cirrus/git_tools.py]: This function is unchanged, just moved github_tools -> git_tools

File: src/cirrus/git_tools.py [hunk 2]
```diff
@@ -200,3 +203,145 @@ def get_diff_files(repo_dir):
...
+        )
+        result.extend(
+            u' --- {0}: {1}'.format(commit['date'],commit['message'])
+            for commit in sorted_commits
+        )
+
+    return '\n'.join(result)
+
+
+def markdown_format(rows):
```
Comment (appeltel) [inline src/cirrus/git_tools.py]: This function is unchanged, just moved github_tools -> git_tools

File: src/cirrus/git_tools.py [hunk 3]
```diff
@@ -200,3 +203,145 @@ def get_diff_files(repo_dir):
...
+
+    return '\n'.join(result)
+
+FORMATTERS = {
+    'plaintext': format_commit_messages,
+    'markdown': markdown_format,
+    }
+
+
+def build_release_notes(repo_dir, since_tag, formatter):
```
Comment (appeltel) [inline src/cirrus/git_tools.py]: This function is unchanged, just moved github_tools -> git_tools

File: tests/unit/cirrus/git_tools_test.py [hunk 4]
```diff
@@ -71,5 +108,72 @@ def test_get_diff_files(self):
...
+                mock_get_tags_sha.return_value = tag
+                mock_get_commit.return_value = self.commit_info
+                build_release_notes(
+                    None,
+                    self.release,
+                    'plaintext')
+                self.failUnless(mock_get_tags_sha.called)
+                self.failUnless(mock_get_commit.called)
+
+    def test_commit_messages(self):
```
Comment (appeltel) [inline tests/unit/cirrus/git_tools_test.py]: This function is unchanged, just moved github_tools -> git_tools

File: tests/unit/cirrus/git_tools_test.py [hunk 5]
```diff
@@ -71,5 +108,72 @@ def test_get_diff_files(self):
...
+        """
+        _test_commit_messages_
+
+        prints plaintext release notes
+        """
+        msg = format_commit_messages(self.commit_info)
+        print "Plaintext release notes:\n{0}\n".format(msg)
+
+    def test_markdown_format(self):
```
Comment (appeltel) [inline tests/unit/cirrus/git_tools_test.py]: This function is unchanged, just moved github_tools -> git_tools

File: tests/unit/cirrus/git_tools_test.py [hunk 6]
```diff
@@ -71,5 +108,72 @@ def test_get_diff_files(self):
         self.failUnlessEqual(diffs[0], path)
         self.failUnless(self.mock_git.Repo.called)
 
+    def test_build_release_notes(self):
```
Comment (appeltel) [inline tests/unit/cirrus/git_tools_test.py]: This function is unchanged, just moved github_tools -> git_tools

File: src/cirrus/git_tools.py [hunk 7]
```diff
@@ -200,3 +203,145 @@ def get_diff_files(repo_dir):
...
+    """
+    _get_tags_
+
+    returns a list of tags for the given repo, ordered as
+    newest first
+
+    """
+    repo = git.Repo(repo_dir)
+    tags_with_date = {
+        tag.name : tag.commit.committed_date
```
Comment (shudgston) [inline src/cirrus/git_tools.py]: extra space before `:`

**Comment by evansde77** [comment]
+1

**Pull request merged**